modules/samples/sample-clients/apim-event-client/src/main/java/org/wso2/carbon/sample/apimevent/Client.java

```diff
@@ -0,0 +1,288 @@
+            "Damro"};
+    private static String[] apiCreator = {"admin", "rohan", "Jeevan"};
+    private static String[] apiMethod = {"GET", "POST", "PUT"};
+    private static String[] applicationId = {"1", "2", "3"};
+    private static String[] applicationName = {"default", "app1", "app2"};
+    private static String[] applicationCK = {"default", "app1", "app2"};
+    private static String[] applicationOwner = {"Michael", "clarke", "cook"};
+    private static String[] userIp = {"10.100.8.14", "10.100.8.26", "10.100.3.64"};
+
+    public static void main(String[] args) {
+
+        DataPublisherUtil.setKeyStoreParams();
+        DataPublisherUtil.setTrustStoreParams();
+
+        log.info("These are the provided configurations: " + Arrays.deepToString(args));
+
+        String protocol = args[0];
+        String host = args[1];
+        String port = args[2];
+        int sslPort = Integer.parseInt(port) + 100;
+        String username = args[3];
+        String password = args[4];
+        String numberOfEventsStr = args[5];
+        int numberOfEvents = Integer.parseInt(numberOfEventsStr);
+
+        try {
+            log.info("Starting APIM Event Client");
+            AgentHolder.setConfigPath(DataPublisherUtil.getDataAgentConfigPath(agentConfigFileName));
+            DataPublisher dataPublisher = new DataPublisher(protocol, "tcp://" + host + ":" + port,
+                    "ssl://" + host + ":" + sslPort, username, password);
+            Event event = new Event();
+            event.setStreamId(DataBridgeCommonsUtils.generateStreamId(STREAM_NAME, VERSION));
+            event.setCorrelationData(null);
+            Event faultEvent = new Event();
+            faultEvent.setStreamId(DataBridgeCommonsUtils.generateStreamId(FAULT_STREAM, VERSION));
+            faultEvent.setCorrelationData(null);
+            Event requestEvent = new Event();
+            requestEvent.setStreamId(DataBridgeCommonsUtils.generateStreamId(REQUEST_STREAM, VERSION));
+            event.setCorrelationData(null);
+
+            String metaClientType = "mozilla";
+
+            for (int i = 0; i < numberOfEvents; i++) {
+                event.setMetaData(new Object[]{metaClientType});
+                faultEvent.setMetaData(new Object[]{metaClientType});
+                requestEvent.setMetaData(new Object[]{metaClientType});
+                Object[] data = getObject();
+                Object[] faultData = getFaultStream();
+                Object[] requestData = getRequestStream();
+                event.setPayloadData(data);
+                faultEvent.setPayloadData(faultData);
+                requestEvent.setPayloadData(requestData);
+                dataPublisher.publish(event);
+                dataPublisher.publish(faultEvent);
+                dataPublisher.publish(requestEvent);
+            }
+            try {
+                Thread.sleep(5000);
+            } catch (InterruptedException e) {
+                log.error(e);
+            }
```

Comment:
please add meaning full error message.

Reply:
Fixed in 1f69fe0038ac7ea895a7fae7e7714add8fc82281